### PR TITLE
More report fixup

### LINKED ISSUE #1367: reports: allow user entered info in title of audit_errors report and some general touchups
### General Touchups:
- [x] status_details (with an s) for report name and rake task, please
- [x] colons in timestamp are awkward in filename on linux box.  remove or change to hyphen or something (just a `gsub` after iso8601 call on DateTime)
- [ ] ~~can we get rid of trailing comma on each data line easily?~~
- [x] allow user entered string for easier file name differentiation between runs (see below)
- [x] remaining feedback from #1365 that wasn't addressed as part of that PR's revisions (see comments on that PR linking to this issue)

### Current title of output file:

`MoabStorageRoot_fixture_sr1_audit_errors_2020-02-12T21:57:42Z.csv`

Breaking this down:
`MoabStorageRoot` - constant for all reports;  dead space as far as I personally am concerned
`fixture_sr1` - name of moab storage root - awesome, keep
`audit_errors` - what kind of report (audit_errors, druids, druid_details) - awesome, keep
`2020-02-12T21:57:42Z` - timestamp - awesome, keep.

What I would like instead:

`fixture_sr1_audit_errors_myTextHere_2020-02-12T21:57:42Z.csv`

That is, i want to provide an argument in the rake task like this:

```
rake prescat:reports:msr_audit_errors[fixture_sr1, naomi-after-lunch]
```

in order to get this:

`fixture_sr1_audit_errors_naomi-after-lunch_2020-02-12T21:57:42Z.csv`

Why?  because the error details in the db after M2C, C2M and CV checks differ, and I want to easily be able to indicate which ones I will be looking at.

TLDR; 

Examples:

after M2C:
```
druid,previous storage root,current storage root,last checksum validation,last moab validation,status,status details
bj102hs9687,,fixture_sr1,2020-02-12 01:10:56 UTC,2018-07-10 17:05:58 UTC,invalid_checksum,validate_checksums (actual location: fixture_sr1; ) Moab file spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0001/data/metadata/versionMetadata.xml was not found in Moab signature catalog spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0003/manifests/signatureCatalog.xml
bj102hs9688,,fixture_sr1,2019-12-13 12:55:59 UTC,2019-12-13 12:31:41 UTC,unexpected_version_on_storage,
```

status of bj102hs9688 is `unexpected_version_on_storage`, no status details

after C2M:
```
druid,previous storage root,current storage root,last checksum validation,last moab validation,status,status details
bj102hs9687,,fixture_sr1,2020-02-12 01:10:56 UTC,2018-07-10 17:05:58 UTC,invalid_checksum,validate_checksums (actual location: fixture_sr1; ) Moab file spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0001/data/metadata/versionMetadata.xml was not found in Moab signature catalog spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0003/manifests/signatureCatalog.xml
bj102hs9688,,fixture_sr1,2019-12-13 12:55:59 UTC,2019-12-13 12:31:41 UTC,online_moab_not_found,check_catalog_version (actual location: fixture_sr1; actual version: 0) db CompleteMoab (created 2019-12-13T12:31:41Z; last updated 2019-12-13T12:55:59Z) exists but Moab not found && CompleteMoab status changed from unexpected_version_on_storage to online_moab_not_found
```

status of bj102hs9688 is `online_moab_not_found`, with status details

after CV:
```
druid,previous storage root,current storage root,last checksum validation,last moab validation,status,status details
bj102hs9687,,fixture_sr1,2020-02-12 22:04:24 UTC,2018-07-10 17:05:58 UTC,invalid_checksum,validate_checksums (actual location: fixture_sr1; ) Moab file spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0001/data/metadata/versionMetadata.xml was not found in Moab signature catalog spec/fixtures/storage_root01/sdr2objects/bj/102/hs/9687/bj102hs9687/v0003/manifests/signatureCatalog.xml
bj102hs9688,,fixture_sr1,2019-12-13 12:55:59 UTC,2019-12-13 12:31:41 UTC,online_moab_not_found,check_catalog_version (actual location: fixture_sr1; actual version: 0) db CompleteMoab (created 2019-12-13T12:31:41Z; last updated 2019-12-13T12:55:59Z) exists but Moab not found && CompleteMoab status changed from unexpected_version_on_storage to online_moab_not_found
```

bj102hs9688 - same as C2M in this example ...

### PR DESCRIPTION
**Pull request by jmartin-sul**
## Why was this change made?

closes #1367

* better rake task names
* remove colons in default output file names
* shorten default output file names
  * i kept `storage_` as a prefix because i'd prefer _some_ namespacing/grouping in the event that we add other reports.  easy enough to totally remove if you feel strongly that it should go.
* user entered "tag" for easier visual differentiation of report runs
* slop from #1365

## Was the usage documentation (e.g. wiki, README, queue or DB specific README) updated?

will update in a subsequent PR, since the changes i had on that front need a little tweaking, and may conflict with changes in (as-yet unmerged) #1372.

### CODE REVIEW

File: app/services/moab_storage_root_reporter.rb [hunk 1]
```diff
@@ -5,11 +5,11 @@
 ##
 # run queries and produce reports from the results, for consumption
 # by preservation catalog maintainers
-class Reporter
+class MoabStorageRootReporter
```
Comment (ndushay) [inline app/services/moab_storage_root_reporter.rb]: yay!

File: app/services/moab_storage_root_reporter.rb [hunk 2]
```diff
@@ -36,21 +36,26 @@ def moab_detail_csv_list(errors_only: false)
...
-    header_row = [
-      ['druid', 'previous storage root', 'current storage root', 'last checksum validation', 'last moab validation', 'status', 'status details']
-    ]
+    header_row = ['druid',
+                  'previous storage root',
+                  'current storage root',
+                  'last checksum validation',
+                  'last moab validation',
+                  'status',
+                  'status details']
```
Comment (ndushay) [inline app/services/moab_storage_root_reporter.rb]: yay!

File: app/services/moab_storage_root_reporter.rb [hunk 3]
```diff
@@ -36,21 +36,26 @@ def moab_detail_csv_list(errors_only: false)
...
 
-    # cols doesn't include storage_root name (available via storage_root ivar).  also we're not instantiating AR objects, so we have to translate
-    # the underlying status enum
+    # cols doesn't include storage_root name (available via storage_root ivar)
```
Comment (ndushay) [inline app/services/moab_storage_root_reporter.rb]: meh - i'd leave this comment out.

File: app/services/moab_storage_root_reporter.rb [hunk 4]
```diff
@@ -59,10 +64,10 @@ def moab_detail_csv_list(errors_only: false)
   # @return [String] the name of the CSV file to which the list was written
   # @raise [ArgumentError] if neither report_type nor filename is provided
   # @raise [RuntimeError] if the file to be written to already exists
-  def write_to_csv(lines, report_type: nil, filename: nil)
+  def write_to_csv(lines, report_type: nil, report_tag: nil, filename: nil)
```
Comment (mjgiarlo) [inline app/services/moab_storage_root_reporter.rb]: @jmartin-sul Can you add YARD documentation for the new param? (I know this has already been merged...)

File: app/services/moab_storage_root_reporter.rb (hunk 3 above)
Comment (jmartin-sul) [reply app/services/moab_storage_root_reporter.rb]: sure, see https://github.com/sul-dlss/preservation_catalog/pull/1374

File: app/services/moab_storage_root_reporter.rb (hunk 4 above)
Comment (jmartin-sul) [reply app/services/moab_storage_root_reporter.rb]: d'oh, sorry, i am just not in the habit of doing YARD doc, see https://github.com/sul-dlss/preservation_catalog/pull/1374
